docs/marketplace/tutorials/fully-functioning-rag-application.md

```diff
@@ -0,0 +1,172 @@
+
+By doing this, whenever our chatbot receives a question, we can:
+
+- Generate embeddings of our question.
+- Search the database for documents with embeddings that are similar in meaning to the embeddings of the question.
+- Retrieve the matched documents and use their text to ensure that the chatbot provides an appropriate answer.
+
+OpenAI has several [embedding models](https://platform.openai.com/docs/guides/embeddings/embedding-models), and MongoDB proposes the [Vector Search Index](https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-overview/) to efficiently search in a database for documents where embeddings are similar in meaning to our question.
+The embeddings from the internal documentation can be created only once, or whenever there is an important update. There are several methods to execute this process. If the documentation is available on a website, the most common and straightforward approach is to perform [web scraping](https://en.wikipedia.org/wiki/Web_scraping). This involves downloading all the pages of the website, removing any unnecessary information such as HTML tags, headers, footers, and styles, and generating embeddings from the remaining text.
+
+The **AI RAG Template** is a perfect example of a tool that can perform all these operations. It provides automatic connections to the OpenAI servers and the MongoDB database, allowing for seamless handling of Vector Indexes. Additionally, it can generate embeddings by simply providing a URL. With this template, creating a RAG application can be done in just a few minutes.
+
+## 1. Install the required applications
+
+We start a new project by creating two applications: the **API Documentation Aggregator** and the **AI RAG Chat**.
+
+To begin, navigate to the design section and click on _Applications_ in the sidebar. Then, select _Create new Application_.
+
+First, search for and select the **API Documentation Aggregator**. This will install the [_API Gateway_](/runtime_suite/api-gateway/10_overview.md) and the [_Swagger Aggregator_](/runtime_suite/swagger-aggregator/10_overview.md), along with several endpoints to expose the Swagger UI for viewing the APIs exposed by the configured services and retrieving the OpenAPI compliant schema.
+
+:::note
+The [API Documentation Aggregator](/runtime_suite_applications/api-documentation-aggregator/10_overview.md) is not required for the chatbot to function, but it is recommended for using the _AI RAG Template_ APIs to generate the embeddings. You can also refer to the instructions [in this tutorial](/console/tutorials/configure-marketplace-components/api-portal.mdx).
+:::
+
+Next, proceed to set up the **AI RAG Chat**. It will prompt you to create the _API Gateway_ (using the existing one with the available listener), the _AI RAG Template_, and the _AI RAG Template Chat_, along with several endpoints to expose the service APIs for sending questions and generating embeddings.
```

Comment:
```suggestion
Next, proceed to set up the **AI RAG Chat**. It will prompt you to create the _API Gateway_ (you can use the previously created one with the available listener), the _AI RAG Template_, and the _AI RAG Template Chat_, along with several endpoints to expose the service APIs for sending questions and generating embeddings.
```